### ISSUE TITLE
Using RequireJS causes error

### ISSUE THREAD

**Issue by thany** [jsmreese/moment-duration-format#57]
To support RequireJS, you're doing this:

```
if (typeof require === "function") {
    try { moment = require('moment'); }
    catch (e) { }
}
```

And that won't work. It's not the correct way to support RequireJS. It spews out this error:

```
Error: Module name "moment" has not been loaded yet for context: _. Use require([])
```

And I'm not even using underscore nor lodash. That's strange, but, the problem is the `require()` call. Calling require like this is not how you call RequireJS, according to the documentation:
http://requirejs.org/docs/errors.html#notloaded

Ultimately, the result of this problem is:

```
uncaught exception: Moment Duration Format cannot find Moment.js
```

Because for RequireJS's gas factory reasons, Moment hadn't been loaded at that time.

TL;DR
Either use the `define` function and put `require()` inside it, or use the async `require([ ])` version.

**Comment by jsmreese** [comment]
Check out the `dev` branch. That was fixed a long time ago but I've never gotten around to publishing an update.
